docTest for ``7-base_geometry.py`` module
=========================================

Using the BaseGeometry class:
-----------------------------

    >>> BaseGeometry = __import__('7-base_geometry').BaseGeometry

TestCases:
----------

    >>> bg = BaseGeometry()

    >>> bg.integer_validator("my_int", 12)
    >>> bg.integer_validator("width", 89)
    >>> bg.integer_validator("pos_int", 42)
    >>> bg.integer_validator("long", 2147483642147483647789)
    >>> bg.integer_validator("mul", 5*5)
    >>> bg.integer_validator("sum", 5+5)
    >>> bg.integer_validator("sub", 5-2)
    >>> bg.integer_validator("", 5)
    >>> bg.integer_validator(None, 5)

    >>> try:
    ...     bg.integer_validator("name", "John")
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [TypeError] name must be an integer

    >>> try:
    ...     bg.integer_validator("intStr", "123")
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [TypeError] intStr must be an integer

    >>> try:
    ...     bg.integer_validator("none", None)
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [TypeError] none must be an integer

    >>> try:
    ...     bg.integer_validator("age", 0)
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [ValueError] age must be greater than 0

    >>> try:
    ...     bg.integer_validator("sub", 5-5)
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [ValueError] sub must be greater than 0

    >>> try:
    ...     bg.integer_validator("age", -2147483642147483647789)
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [ValueError] age must be greater than 0

    >>> try:
    ...     bg.integer_validator("distance", -4)
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [ValueError] distance must be greater than 0

    >>> try:
    ...     bg.integer_validator("array", [5, 6])
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [TypeError] array must be an integer

    >>> try:
    ...     bg.integer_validator("set", {1, 2, 3})
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [TypeError] set must be an integer

    >>> try:
    ...     bg.integer_validator("float", 5.5)
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [TypeError] float must be an integer

    >>> try:
    ...     bg.integer_validator("subFloat", 5-2.5)
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [TypeError] subFloat must be an integer

    >>> try:
    ...     bg.integer_validator("div", 5/2)
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [TypeError] div must be an integer

    >>> try:
    ...     bg.integer_validator("bool", True)
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [TypeError] bool must be an integer

    >>> try:
    ...     bg.integer_validator("dict", {'1': 1, '2': 2})
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [TypeError] dict must be an integer

    >>> try:
    ...     bg.integer_validator("tuple", (1, 3))
    ... except Exception as e:
    ...     print("[{}] {}".format(e.__class__.__name__, e))
    [TypeError] tuple must be an integer
